src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -142,6 +142,33 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 telemetryEventDetails.SetProperty(TelemetryConstants.PopToken, authenticationResult.TokenType.Equals(Constants.PoPTokenType));
                 telemetryEventDetails.SetProperty(TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds);
                 telemetryEventDetails.SetProperty(TelemetryConstants.ActivityId, authenticationResult.CorrelationId);
+                telemetryEventDetails.SetProperty(TelemetryConstants.RefreshOn, 
+                    authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue ?
+                    DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value)
+                    : 0);
+
+                LogCacheDetailsToTelemetryClient(cacheDetails, telemetryEventDetails);
+            }
+        }
+
+        private void LogCacheDetailsToTelemetryClient(Dictionary<string, object> cacheDetails, MsalTelemetryEventDetails telemetryEventDetails)
+        {
+            foreach (var cacheDetail in cacheDetails)
+            {
+                switch (cacheDetail.Key)
+                {
+                    case TelemetryCacheConstants.CacheUsed:
+                        telemetryEventDetails.SetProperty(TelemetryCacheConstants.CacheUsed, ((CacheUsed)cacheDetail.Value).ToString());
+                        break;
+                    case TelemetryCacheConstants.L1Latency:
+                        telemetryEventDetails.SetProperty(TelemetryCacheConstants.L1Latency, (long)cacheDetail.Value);
```

Comment:
I think you need to sanitize the values, there is no guarantee that those objects are ints.